src/libraries/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryFormatterWriter.cs

```diff
@@ -268,7 +268,7 @@ private void WriteArrayAsBytes(Array array, int typeLength)
                 if (!BitConverter.IsLittleEndian)
                 {
                     // we know that we are writing a primitive type, so just do a simple swap
-                    Debug.Fail("Re-review this code if/when we start running on big endian systems");
+                    // Debug.Fail("Re-review this code if/when we start running on big endian systems");
```

Comment:
Just delete it instead of commenting it out

Reply:
Pushed amendment